### ISSUE TITLE
Vyper ignores unnecessary keyword arguments

### ISSUE THREAD

**Issue by robinsierra** [ethereum/vyper#1554]
### Version Information

* vyper version: 0.1.0b10
* OS: osx

### What's your issue about?

When calling a function Vyper ignores unnecessary keyword arguments, e.g.
```
@public
def foo():
   pass

@public
def bar():
   self.foo(kw=1234)
```
compiles without throwing an error. This is especially surprising when trying to call a payable function with a value
```
@public
@payable
def foo():
   pass

@public
def bar():
   self.foo(value=self.balance)
```
as the value argument will simply be ignored.

### How can it be fixed?

Disallow using keyword arguments where none are required.

**Comment by charles-cooper** [comment]
@robinsierra how do you know the `value` argument is ignored?

**Comment by robinsierra** [comment]
When running the following contract in remix 

```
amount: wei_value

@public
def get_balance() -> wei_value:
    return self.balance

@public
def get_amount() -> wei_value:
    return self.amount

@public
@payable
def pay():
    pass

@private
def get_and_set_amount() -> wei_value:
    self.amount = self.amount + 1
    return self.amount

@public
def send_money():
    self.pay(value=self.get_and_set_amount())
```
i.e. calling send_money the call will always succeed, even though `balance == 0`, and `amount` never changes.

**Comment by robinsierra** [comment]
Although that might be because passing an amount to a call was not yet possible in the version that the remix IDE uses.

**Comment by fubuloubu** [comment]
Closed by #1559
